From fa5dc9562fdce3a7781b0fcee0c199f45b0dcdb1 Mon Sep 17 00:00:00 2001
From: Lin Fuzhen <fuzhen.lin@freescale.com>
Date: Thu, 12 Apr 2012 18:38:57 +0800
Subject: [PATCH 200/249] ENGR00179049 mx6vpu: Add vpu support for mx6dl

For MX6Q and MX6Dl use the same registers map, so use BOARD_SOC_CLASS
instead of BOARD_SOC_TYPE to avoid more code changes.

Signed-off-by: Lin Fuzhen <fuzhen.lin@freescale.com>
---
 vpu/Android.mk | 4 ++++
 vpu/vpu_reg.h  | 2 +-
 vpu/vpu_util.h | 2 +-
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/vpu/Android.mk b/vpu/Android.mk
index 0565113..6620784 100755
--- a/vpu/Android.mk
+++ b/vpu/Android.mk
@@ -9,7 +9,11 @@ LOCAL_SRC_FILES := \
 	vpu_util.c \
 	vpu_lib.c \
 	vpu_gdi.c
+ifeq ($(BOARD_SOC_CLASS),IMX6)
+LOCAL_CFLAGS += -DBUILD_FOR_ANDROID -DIMX6Q
+else
 LOCAL_CFLAGS += -DBUILD_FOR_ANDROID -D$(BOARD_SOC_TYPE)
+endif
 LOCAL_C_INCLUDES += $(LOCAL_PATH)
 LOCAL_SHARED_LIBRARIES := libutils libc liblog
 LOCAL_MODULE := libvpu
diff --git a/vpu/vpu_reg.h b/vpu/vpu_reg.h
index 760bfd3..e0e87bc 100644
--- a/vpu/vpu_reg.h
+++ b/vpu/vpu_reg.h
@@ -396,7 +396,7 @@
         #define TEMP_BUF_SIZE			(304 * 1024)
         #define PARA_BUF2_SIZE			(2 * 1024)
         #define PARA_BUF_SIZE			(10 * 1024)
-#elif defined(IMX6Q) || defined(IMX6DL)
+#elif defined(IMX6Q)
 	#define CODE_BUF_SIZE                   (280 * 1024) /* define max is 280k byte currently */
 	#define FMO_SLICE_SAVE_BUF_SIZE         (32)
 	#define TEMP_BUF_SIZE                   (204 * 1024)
diff --git a/vpu/vpu_util.h b/vpu/vpu_util.h
index 4dfd0e1..d02e89f 100644
--- a/vpu/vpu_util.h
+++ b/vpu/vpu_util.h
@@ -100,7 +100,7 @@ enum {
 	AVS_DEC = 0x81,
 	VPX_DEC = 0x82
 } ;
-#elif defined(IMX6Q) || defined(IMX6DL)
+#elif defined(IMX6Q)
 
 enum {
 	AVC_DEC = 0,
-- 
1.8.0

